xarray/core/coordinates.py

```diff
@@ -228,17 +276,40 @@ def __init__(
         from xarray.core.dataset import Dataset
 
         if coords is None:
-            variables = {}
-        elif isinstance(coords, Coordinates):
+            coords = {}
+
+        variables: dict[Hashable, Variable]
+        default_indexes: dict[Hashable, PandasIndex] = {}
+        coords_obj_indexes: dict[Hashable, Index] = {}
+
+        if isinstance(coords, Coordinates):
+            if indexes is not None:
+                raise ValueError(
+                    "passing both a ``Coordinates`` object and a mapping of indexes "
+                    "to ``Coordinates.__init__`` is not allowed "
+                    "(this constructor does not support merging them)"
+                )
             variables = {k: v.copy() for k, v in coords.variables.items()}
+            coords_obj_indexes = dict(coords.xindexes)
         else:
-            variables = {k: as_variable(v) for k, v in coords.items()}
+            variables = {}
+            for name, data in coords.items():
+                var = as_variable(data, name=name)
```

Comment:
This still converts any coordinate input in the form `{"x": array_like}` to an `IndexVariable`, which triggers data loading by coercing `array_like` to a `pd.Index`.

It will be fixed in another PR (#8124).

As a workaround, `{"x": ("x", array_like)}` does not coerce `array_like`.

Reply:
@benbovy I'm pretty sure this workaround currently does not work. I tried to create a `Coordinates` object from scratch with no indexes and the only way I was able to do it required reverting xarray to `v2023.08.0`, i.e. just before this PR was merged.

See the end of https://gist.github.com/TomNicholas/d9eb8ac81d3fd214a23b5e921dbd72b7 for full context.

Reply:
Raised with MCVE in https://github.com/pydata/xarray/issues/8704